File: 2/JS/main.js
function fibonacci(x) {
	if (x <= 1) {
		return x;
	} else {
		return fibonacci(x-1) + fibonacci(x-2);
	}
}

function solve() {
	var sum = 0;
	var value = 0;
	var counter = 0;

	while (value < 4000000){
		value = fibonacci(counter);
		if (value % 2 == 0){
			sum += value;
		}
		counter++;
	}
	return sum;
}

console.log(solve());

File: 2/python/main.py
def fibonacci(x) -> int :
    if x <= 1 :
        return x
    else :
        return fibonacci(x-1) + fibonacci(x-2)

def solve() -> int :
    sum = 0
    value = 0
    counter = 0
    while value < 4000000:
        value = fibonacci(counter)
        if value % 2 == 0 :
            sum += value
        counter += 1
    return sum

def main() -> None :
    print(solve())

if __name__ == "__main__":
    main()
